### PR TITLE
Interclip v4

### PR DESCRIPTION
**Pull request by filiptronicek**
Wow! It's been 2 years since Interclip v3, so I figured it's time for v4! Here are some things I have prepared for v4:

## Features
- [x] ~~New hamburger menu~~ **Horizontal menu on mobile**
- [x] Native HTML input validation for clip codes (https://regexr.com/61gte)
- [x] A section on the homepage for viewing recently made clips
- [x] Completely new drag and drop interface 
- [x] A feature to copy the newly made code

## Design
- [ ] **New logoo!**
- [x] Redesigned receive page (again, because @kahy9 doesn't like it)
- [x] Redesigned about page
- [x] Redesigned home page
- [x] Redesigned privacy policy page
- [x] Redesigned admin dashboard
- [x] **New design ([Figma](https://www.figma.com/file/ACRahTRI2S6VQbw0APvoHU/Interclip.app?node-id=88%3A39))**

## Code and performance
- [x] A router for delegating requests to the server (#111)
- [x] Add Redis for caching (LRU cache: https://redis.io/topics/lru-cache)
- [x] More semantic HTML, that makes more sense
- [x] A big codebase refactor

## Docs
- [x]  Update the docs (now at https://docs.interclip.app/)
- [ ] Update the social image on GitHub
- [ ] Update the icons on Product Hunt, Taskord, and the mobile app

## Known issues and bugs
- The admin dashboard has a lot of broken styles

## Other
- [ ] Move the repo to interclip/interclip

## API
The API will stay mostly the same, but the endpoints will be updated.
The new clip endpoints will be `/api/set` and the receive clip endpoint will be `/api/get`, this is the second major endpoint change since Interclip had an API, and if you're wondering when Interclip got its API, it was on September 5th, 2019 in f7ff33bf432377bbc4d9f43d191d6b5a7b1786a3

### CODE REVIEW

**Review by bigint — COMMENTED** [review]
Typo in commit 😅

**Comment by filiptronicek** [comment]
> 
> 
> Typo in commit 😅

Yup happens 😅

File: css/index.css [hunk 1]
```diff
@@ -179,7 +153,21 @@ input::placeholder {
...
+  -webkit-flex-direction: column;
+     -moz-box-orient: vertical;
+     -moz-box-direction: normal;
+      -ms-flex-direction: column;
+          flex-direction: column;
+  -webkit-box-align: center;
+  -webkit-align-items: center;
+     -moz-box-align: center;
+      -ms-flex-align: center;
+          align-items: center;
```
Comment (bigint) [inline css/index.css:172]: This is not aligned :D

File: css/menu.css [hunk 2]
```diff
@@ -110,17 +142,16 @@ li a:hover:not(.active) {
...
 
   .lg {
     display: none;
   }
 }
-
 /* The Close Button */
 .closeBtn {
-  color: #aaaaaa;
+  color: #aaa;
```
Comment (bigint) [inline css/menu.css:154]: 🤔 I think the standard is to have 6 chars (HEX)

File: css/menu.css [hunk 3]
```diff
@@ -156,47 +195,47 @@ li a:hover:not(.active) {
...
-  cursor: pointer;
+  -webkit-border-radius: 0.25em;
+     -moz-border-radius: 0.25em;
+          border-radius: 0.25em;
 }
 
 /* Arrow */
 .select::after {
-  content: "\25BC";
+  content: "▼";
```
Comment (bigint) [inline css/menu.css:216]: I suggest using icon for this instead of symbols

File: css/menu.css [hunk 4]
```diff
@@ -307,19 +383,20 @@ h1:nth-child(1) {
...
+#adminbar:focus {
   display: inline-block;
 }
 
 .skip-link {
   position: absolute;
   top: -100px;
   left: 0;
-  background: #000000;
+  background: #000;
```
Comment (bigint) [inline css/menu.css:399]: HEX 😅

File: scss/_mvp_reverts.scss [hunk 5]
```diff
@@ -0,0 +1,11 @@
+a {
+    font-weight: unset;
```
Comment (bigint) [inline scss/_mvp_reverts.scss:2]: Try using 2 spaces cuz you used 2 in `scss/_colors.scss`

File: css/menu.css (hunk 2 above)
Comment (filiptronicek) [reply css/menu.css:138]: Unfortunately, this is due to Sass doing that, the un-compiled version does not have it.

File: includes/menu.php [hunk 6]
```diff
@@ -11,8 +11,8 @@
   'privacy' => ['privacy', 'Privacy policy'],
 );
 
-if ($isStaff) {
-  $pages += [ 'admin' => ['admin', 'Admin'] ];
+if ($isStaff && $_ENV['ENVIRONMENT'] !== 'staging') {
```
[the remark this answers is not in the document]
Comment (bigint) [reply includes/menu.php:14]: ☝🏻

File: includes/components/html/new/ok-msg.php [hunk 7]
```diff
@@ -78,4 +78,22 @@ function handleChange() {
...
+        if (recentlyMadeArray.length > 6) {
+            const reversedRecents = recentlyMadeArray.reverse();
+            reversedRecents.pop();
+            localStorage.setItem("recentClips", JSON.stringify(reversedRecents.reverse()));
+        } else {
+            localStorage.setItem("recentClips", JSON.stringify(recentlyMadeArray));
+        }
+    } else {
+        localStorage.setItem("recentClips", JSON.stringify([url]))
+    }
```
Comment (bigint) [inline includes/components/html/new/ok-msg.php:98]: No return statements 🤔

File: js/index.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+const getEntries = () => {
+    const initialValue = localStorage.getItem("recentClips");
+
+    if (initialValue) {
+        return JSON.parse(initialValue);
+    } else {
+        return [];
+    }
```
Comment (bigint) [inline js/index.js:22]: Else is not needed when using return in `if` 😄

**Review by bigint — CHANGES_REQUESTED** [review]

File: includes/components/html/new/ok-msg.php (hunk 7 above)
Comment (filiptronicek) [reply includes/components/html/new/ok-msg.php:98]: I don't think there shouldn't be any, or should there? It is not in any function, so it cannot return

File: js/file.js [hunk 9]
```diff
@@ -42,14 +42,21 @@ function uploadRe(file) {
     progressBar.style.display = "none";
     progressValue.innerText = "Uploading to IPFS....";
 
+    let providerEndpoint = "https://cloudflare-ipfs.com";
+
+    if (file.type.match(new RegExp("video\/.{1,10}"))) {
```
Comment (bigint) [inline js/file.js:47]: This is good 👏🏻

Comment (filiptronicek) [reply js/file.js:47]: Thanks! <3

File: site.webmanifest [hunk 10]
```diff
@@ -15,5 +15,12 @@
   ],
   "theme_color": "#ffffff",
   "background_color": "#ffffff",
-  "display": "standalone"
+  "display": "standalone",
+  "prefer_related_applications": true,
+  "related_applications": [
```
Comment (bigint) [inline site.webmanifest:20]: Learned a new thing 🤩

**Comment by filiptronicek** [comment]
Here goes nothing, but this is for you, @bigint, happy birthday!

**Pull request merged**